pkg/controllers/routereflector/datastores/etcd.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 IBM Corporation All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package datastores
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/projectcalico/libcalico-go/lib/apiconfig"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/options"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type calicoNodeClient interface {
```

Comment:
same question here - why do we need a special interface for this?

Reply:
same answer here. There are not too many tests at the moment, but I'm planning to implement them and with this i can tell what should effects need to be mocked out.